### PR TITLE
AI Assistant: iterate over custom hook to request suggestions

### LINKED ISSUE #31379: Ai Assistant: update block content when `suggestion` event
The block text content should be updated when `suggestion` event happens.

### PR DESCRIPTION
**Pull request by retrofox**
On this PR:

* Fixes a minor issue when setting the block content
* Better handling on subscribing event source events

Fixes https://github.com/Automattic/jetpack/issues/31379

## Proposed changes:

* AI Assistant: iterate over custom hook to request suggestions

### Other information:

- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to the block editor
* Create a paragraph/heading block
* Generate the content. Use `Extend` since you'll need time to do the next step
* Remove the block
* Confirm the client stops requesting.

You can compare the behavior with the AI Assistant block. It keeps requesting even when it's removed from the editor.

* Test the rest of the options. Confirm it keeps working as expected.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-ai/index.ts [hunk 1]
```diff
@@ -83,7 +78,24 @@ export default function useSuggestionsFromAI( {
...
+		[ onSuggestion ]
+	);
+
+	/**
+	 * onDone function handler.
+	 *
+	 * @param {string} content - The content.
+	 * @returns {void}
+	 */
+	const handleDone = useCallback( ( event: CustomEvent ) => onDone( event?.detail ), [ onDone ] );
```
Comment (dhasilva) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-ai/index.ts:98]:
```suggestion
	const handleDone = useCallback( ( event: CustomEvent ) => onDone?.( event?.detail ), [ onDone ] );
```

`onDone` is not required

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-ai/index.ts [hunk 2]
```diff
@@ -83,7 +78,24 @@ export default function useSuggestionsFromAI( {
...
 
-	const readyToRequest = postId && prompt?.length;
+	/**
+	 * onSuggestion function handler.
+	 *
+	 * @param {string} suggestion - The suggestion.
+	 * @returns {void}
+	 */
+	const handleSuggestion = useCallback(
+		( event: CustomEvent ) => onSuggestion( event?.detail ),
```
Comment (dhasilva) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-ai/index.ts:88]:
```suggestion
		( event: CustomEvent ) => onSuggestion?.( event?.detail ),
```

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-ai/index.ts (hunk 1 above)
Comment (dhasilva) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-ai/index.ts:98]: Ah, just saw that it is only added as listener callback if `onDone` exists.

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-ai/index.ts (hunk 2 above)
Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-ai/index.ts:88]: same, right?

**Review by dhasilva — APPROVED** [review]
LGTM

**Pull request merged**